mirrord/agent/iptables/src/lib.rs

```diff
@@ -238,6 +217,28 @@ where
         Ok(Self { redirect })
     }
 
+    /// detect if the IP tables on the target are dirty: this is required since we use static
+    /// IP tables chain names, and they are in danger of conflicting with each-other
+    ///
+    /// returns `true` if iptables are clean and the agent can proceed, other returns `false`
+    #[tracing::instrument(level = Level::TRACE, skip(ipt) ret, err)]
+    pub async fn ensure_iptables_clean(ipt: IPT) -> IPTablesResult<bool> {
```

Comment:
```suggestion
    pub async fn list_mirrord_rules(ipt: IPT) -> IPTablesResult<Vec<String>> {
```
I'd make it return the list of rules that mention mirrord chains. Then, in the caller function, you can do a nice error log:
```rust
let leftover_rules = SafeIpTables::list_mirrord_rules(ipt).await?;
if existing_mirrord_rules.is_empty().not() {
    tracing::error!(
        leftover_rules = ?leftover_rules,
        "Detected dirty iptables. Either some other mirrord agent is running \
        or the previous agent failed to clean up before exit. \
        If no other mirrord agent is targeting this pod, please delete the pod."
    );
    let _ = notify_client_about_dirty_iptables(args).await; // Try to create the tcp listener, accept one connection, and send DaemonMessage::Close.
    return Err(AgentError::DirtyIpTables);
}
```